fluent/client/ws_client.go

```diff
@@ -114,12 +112,12 @@ func (wcf *DefaultWSConnectionFactory) New() (ext.Conn, error) {
 	}
 
 	conn, resp, err := dialer.Dial(wcf.URL, header)
+
```

Comment:
We should check `err` before handling `resp`, e.g:

```
conn, resp, err := wcf.Dialer.Dial(wcf.URL, header)
if err != nil {
	return nil, err
}

```

Reply:
I have initially tried this but most of times when we receive `err`, we also received response code like resp (which contains error response code)

Reply:
Ok, good to know, probably the resp status code is more informative than error we get from ws library.